pkg/apis/ocs/v1/topologymap.go

```diff
@@ -0,0 +1,44 @@
+package v1
+
+import (
+	"strings"
+)
+
+// Contains checks whether the NodeTopologyMap contains a specific value
+// for the specified key
+func (m *NodeTopologyMap) Contains(topologyKey string, value string) bool {
+	if values, ok := m.Labels[topologyKey]; ok {
+		if _, ok := values[value]; ok {
+			return values[value]
+		}
+	}
+
+	return false
+}
+
+// Add adds a new value to the NodeTopologyMap under the specified key
+func (m *NodeTopologyMap) Add(topologyKey string, value string) {
+	if _, ok := m.Labels[topologyKey]; !ok {
+		m.Labels[topologyKey] = TopologyLabelValues{}
+	}
+
+	m.Labels[topologyKey][value] = true
+}
+
+// GetKeyValues returns a node label matching the topologyKey and all values
+// for that label across all storage nodes
+func (m *NodeTopologyMap) GetKeyValues(topologyKey string) (string, []string) {
```

Comment:
`GetNodeLabelUsingKey` looks understandable and self-explanatory.

Reply:
You're not just getting a label, you are getting all values for all labels matching the specified `topologyKey`.